### PR TITLE
feat: add barrel flux return

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This PR adds the barrel flux return backlegs and spacers per the info in #693. Needed for hall background radiation studies. Also adds forward endcap flux return.

image

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators @ajentsch 

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
Not really for any detector behavior.

### CODE REVIEW

**Comment by wdconinc** [comment]
Currently overlaps with LFHCAL since:
```
LFHCAL_rmax                    =      270.030 = HcalBarrel_rmax // should be 267
HcalBarrel_rmax                =      270.030 = HcalBarrel_rmin1 + HcalBarrel_thickness // should be 268.2
HcalBarrel_rmin1               =      183.850 = 183.85*cm // should 180.0
HcalBarrel_rmin2               =      194.250 = HcalBarrel_rmin1 + 10.4*cm // correct
```

@johnlajoie @ruse-traveler This issue seems to start with HcalBarrel_rmin1. What should be changed here to end up with HcalBarrel_rmax = 268.2 cm?

**Comment by johnlajoie** [comment]
As far as I am concerned, the 268.2 cm in the detector table is incorrect, and I went around and around with the project with this for months.  As a reminder, what is in the ePIC geometry for the outer HCAL is the 3D geometry as-built. Rolf explained to me that the detector parameter table is generated from a 2D drawing, and my best guess is that there is some confusion because what they are looking at is 2D and the plate is tilted at an angle. (This can't be the whole problem or it would be a larger discrepancy. It could also have to do with the fact that I think their 2D drawing has a plate and not the completed sector, which can be different, esp. with the end plates.)    Rolf insisted that all they can work from is the 2D drawing. 

In my last request for an update I provided them with a set of dimensions from the 3D model: 

https://eic.jlab.org/Geometry/Detector/local/H/HCAL_Geometry_Summary_from_3D.pdf

which is linked to the Detector Parameter table. I see now they updated some of the parameters in my request, but not others, without comment. 

I can revisit this again with Walt, Tanja and Rolf if necessary

**Comment by wdconinc** [comment]
Ok, so the barrel hcal is fine, and it's just that the positive endcap flux return dimensions are not in agreement with the CAD model. I'll update the positive endcap flux return, and this should resolve the overlaps.

CAD
table

E.g. inner radius of outer collar highlighted: 269 cm in table, 289.56 cm in drawing.

**Comment by wdconinc** [comment]
FluxEndcapN dimensions
CAD
table

**Comment by wdconinc** [comment]
With 7fcecd0 the LFHCAL is now in an envelope volume that is a tube with off-center box removed. This envelop volume is made of Steel235 to account for the "staircase" pattern in https://github.com/eic/epic/pull/694#issuecomment-2037737708.

**Comment by wdconinc** [comment]
Envelope volume of LFHCAL:
image

**Comment by wdconinc** [comment]
This does not add any spurious extra material to the LFHCAL. The output itself is too long, but no changes in sha256 on a non-trivial ray from the origin.

Before:
```
$ materialScan /opt/detector/epic-nightly/share/epic/epic_lfhcal_only.xml 0 0 0 1 200 500 | sha256sum
cfeeaaaf182d53fc67fedc92ad51a28ee61b122b8d7d770d419c3df06b5c6fee  -
```

After:
```
$ materialScan /home/wdconinc/git/epic/.worktree/693-add-barrel-flux-return-backlegs-spacers/prefix/share/epic/epic_lfhcal_only.xml 0 0 0 1 200 500 | sha256sum
cfeeaaaf182d53fc67fedc92ad51a28ee61b122b8d7d770d419c3df06b5c6fee  -
```

File: compact/definitions.xml [hunk 1]
```diff
@@ -630,17 +630,35 @@ Service gaps in FW direction (before endcapP ECAL) and BW direction (before endc
...
+    <constant name="FluxEndcapP_collar_rmin" value="289.56*cm"/>
+    <constant name="FluxEndcapP_collar_thickness" value="167.14*cm"/>
+    <constant name="FluxEndcapP_oculus_rmax" value="FluxEndcapP_collar_rmin"/>
+    <constant name="FluxEndcapP_oculus_rmin" value="210.0*cm"/>
+    <constant name="FluxEndcapP_oculus_thickness" value="22.2*cm"/>
+
+    <comment> LFHCAL includes support, which is part of the flux return </comment>
+    <constant name="LFHCAL_rmax" value="FluxEndcapP_collar_rmin"/>
+
+    <constant name="FluxBarrelForward_zmax"  value="320*cm"/>
```
Comment (wdconinc) [inline compact/definitions.xml:655]:
```suggestion
    <comment> STAR Asm for EPIC w cradle.stp, rwimmer, 2024-04-03 </comment>
    <constant name="FluxBarrelForward_zmax"  value="320*cm"/>
```

File: compact/hcal/lfhcal.xml [hunk 2]
```diff
@@ -231,7 +236,7 @@
         <position x="40*cm" y="-35*cm" z="0*cm"/>
         <position x="-30*cm" y="-45*cm" z="0*cm"/>
         <position x="30*cm" y="-45*cm" z="0*cm"/>
-        <position x="0*cm" y="-55*cm" z="0*cm"/>
+        <position x="0* cm" y="-55*cm" z="0*cm"/>
```
Comment (wdconinc) [inline compact/hcal/lfhcal.xml:239]:
```suggestion
        <position x="0*cm" y="-55*cm" z="0*cm"/>
```

**Comment by veprbl** [comment]
For some reason endcaps have 50% transparency in teveDisplay. Haven't been able to figure out why.

**Review by veprbl — APPROVED** [review]

**Pull request merged**